src/Orleans.KafkaStreamProvider/KafkaQueue/KafkaStreamProviderOptions.cs

```diff
@@ -164,5 +169,23 @@ private static bool GetOptionalParamBool(string paramName, bool defaultValue, IP
 
             return paramValue;
         }
+        
+        private Func<IKafkaBatchFactory> GetOptionalParamKafkaBatchFactory(string paramName, Func<IKafkaBatchFactory> defaultValue, IProviderConfiguration config, IServiceProvider serviceProvider)
+        {
+            string paramValuePreParsed;
+            var paramValue = defaultValue;
+            if (!config.Properties.TryGetValue(paramName, out paramValuePreParsed)) return paramValue;
+
+            Type batchFactoryType = Type.GetType(paramValuePreParsed, false, true);
+
+            if (batchFactoryType == null)
+                throw new ArgumentException(String.Format("Could not locate the type {0}. Please ensure it is a fully qualified type name and the associated assembly has been loaded", paramName));
+            else if (!typeof(IKafkaBatchFactory).IsAssignableFrom(batchFactoryType))
+                throw new ArgumentException(String.Format("The specified type {0} does not implement IKafkaBatchFactory", paramName));
+            else
+                paramValue = () => (serviceProvider.GetService(batchFactoryType) ?? Activator.CreateInstance(batchFactoryType)) as IKafkaBatchFactory;
```

Comment:
@EranOfer, the fallback is there so that it is not necessary to specifically register your IKafkaBatchFactory implementation with the Orleans ServiceProvider. It's probably preferable that you *do* register it... but I didn't see any harm in supporting both options.